commands/peers/addpeer.go

```diff
@@ -35,6 +35,20 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 		req.Name = req.Addresses[0]
 	}
 
+	local_node := false
+	for _, addr := range req.Addresses {
+		local, err := utils.IsLocalAddress(addr)
+		if err != nil || local == true {
```

Comment:
if err != nil then you should error out with a different code than ErrPeerLocalNode.